MSAL/src/native_auth/controllers/MSALNativeAuthBaseController.swift

```diff
@@ -149,12 +149,20 @@ class MSALNativeAuthBaseController {
         completion(response, error)
     }
 
-    func performRequest<T>(_ request: MSIDHttpRequest, context: MSIDRequestContext) async -> Result<T, Error> {
+    func performRequest<T: MSALNativeAuthResponseCorrelatable>(
+        _ request: MSIDHttpRequest,
+        context: MSALNativeAuthRequestContext
+    ) async -> Result<T, Error> {
         return await withCheckedContinuation { continuation in
             request.send { result, error in
-                if let error = error {
+                if let error = error, let errorWithHeaders = error as? MSALNativeAuthResponseCorrelatable {
```

Comment:
`errorWithHeaders` should be `errorWithCorrelationId` or `correlatableError` if you want to use the new word :)

Reply:
Thank you, done